src/interpreter.ts

```diff
@@ -765,36 +766,28 @@ export class Interpreter {
         );
     }
 
-    public interpretExpression(ast: AstExpression): Value {
-        switch (ast.kind) {
-            case "id":
-                return this.interpretName(ast);
-            case "method_call":
-                return this.interpretMethodCall(ast);
-            case "init_of":
-                return this.interpretInitOf(ast);
-            case "null":
-                return this.interpretNull(ast);
-            case "boolean":
-                return this.interpretBoolean(ast);
-            case "number":
-                return this.interpretNumber(ast);
-            case "string":
-                return this.interpretString(ast);
-            case "op_unary":
-                return this.interpretUnaryOp(ast);
-            case "op_binary":
-                return this.interpretBinaryOp(ast);
-            case "conditional":
-                return this.interpretConditional(ast);
-            case "struct_instance":
-                return this.interpretStructInstance(ast);
-            case "field_access":
-                return this.interpretFieldAccess(ast);
-            case "static_call":
-                return this.interpretStaticCall(ast);
-        }
-    }
+    public interpretExpression = (ast: AstExpression): Value =>
+        match(ast)
+            .on({ kind: "id" })((ast) => this.interpretName(ast))
```

Comment:
Cool, but doesn't this create a ton of small `{ kind: whatever }` objects each time `interpretExpression()` is called?